lib/transaction.c

```diff
@@ -109,6 +121,27 @@ static char *getMntPoint(const char *dirName, dev_t dev)
     return res;
 }
 
+static int getRotational(const char *dirName, dev_t dev)
+{
+    int rotational = 1;	/* Be a good pessimist, assume the worst */
+#if defined(__linux__)
+    char *devpath = NULL;
+    FILE *f = NULL;
+
+    rasprintf(&devpath, "/sys/dev/block/%d:%d/queue/rotational",
+			major(dev), minor(dev));
+    if ((f = fopen(devpath, "r")) != NULL) {
+	int v;
+	if (fscanf(f, "%d", &v) == 1)
+	    rotational = v;
```

Comment:
Should that be `rotational = v ? 1 : 0` or something like that?

Reply:
AIUI the values in that file are always 0 or 1 to begin with, because this is just a simple boolean: is it rotational or not?

Reply:
That maybe the case with current kernels, but with kernel version 6.0.0 it will probably return '-1' for some disks...

I'm just nitpicking because the rest of the code assumes that it returns a boolean, which is not something fscanf guarantees. Feel free to ignore this comment ;-)

Reply:
I don't mind being nitpicked at all, it's a fair point. Fixed now to only consider value of '1' as rotational, because -1 to me sounds like a "I dunno" value.